plugins/analytics-provider-segment/config.d.ts

```diff
@@ -20,26 +20,47 @@ export interface Config {
     // context of the monorepo is too strict. Ideally, this would be marked as
     // required.
     analytics?: {
-      segment: {
-        /**
-         * The Segment write key.
-         * @visibility frontend
-         */
-        writeKey: string;
+      segment:
+        | {
+            /**
+             * The Segment write key.
+             * @visibility frontend
+             */
+            writeKey?: string;
 
-        /**
-         * Prevents events from actually being sent when set to true. Defaults
-         * to false.
-         * @visibility frontend
-         */
-        testMode?: boolean;
+            /**
+             * Prevents events from actually being sent when set to true. Defaults
+             * to false.
+             * @visibility frontend
+             */
+            testMode: true;
```

Comment:
Why can't we leave it `testMode?`

Reply:
If we left it as `testMode?: true` that would intersect with `testMode?: false`. For a complete union that does not intersect we want `testMode: true` and `testMode?: false`. i.e. `true` and `false | undefined` -> `boolean | undefined`.